File: frontend/src/types/Training.ts
// Типы данных для тренировок
export enum TrainingType {
  Pool = 'Бассейн',
  Depth = 'Глубина',
  Gym = 'Зал',
  Other = 'Другое'
}

export interface Exercise {
  id: string;
  name: string;
  notes: string;
}

export interface Training {
  id: string;
  type: TrainingType;
  date: Date;
  difficulty: number; // 1-5
  notes?: string;
  
  // Специфические поля для разных типов тренировок
  poolTraining?: {
    poolSize?: number;
  };
  
  depthTraining?: {
    wetsuit?: number; // толщина костюма в мм
    temperature?: number;
    weights?: number;
    location?: string;
    maxDepth?: number;
  };
  
  gymTraining?: {
    avgHeartRate?: number;
    maxHeartRate?: number;
    calories?: number;
  };
  
  exercises: Exercise
}

File: backend/app/schemas/training.py
from pydantic import BaseModel, Field
from datetime import date
from typing import Optional, List, Literal
from enum import Enum

from app.schemas.exercise import ExerciseResponse

class TrainingType(str, Enum):
    Pool = 'Pool'
    Depth = 'Depth'
    Gym = 'Gym'
    Other = 'Other'

class TrainingBase(BaseModel):
    type: TrainingType
    date: date
    difficulty: int = Field(..., ge=1, le=5)
    notes: Optional[str] = Field(None, max_length=500)

    poolTraining: Optional[dict] = Field(None)
    depthTraining: Optional[dict] = Field(None)
    gymTraining: Optional[dict] = Field(None)

class TrainingCreate(TrainingBase):
    pass

class TrainingUpdate(TrainingBase):
    type: Optional[TrainingType] = None
    date: Optional[str] = None
    difficulty: Optional[int] = Field(None, ge=1, le=5)
    notes: Optional[str] = Field(None, max_length=500)
    poolTraining: Optional[dict] = None
    depthTraining: Optional[dict] = None
    gymTraining: Optional[dict] = None

class TrainingResponse(TrainingBase):
    id: int
    user_id: int

    class Config:
        from_attributes = True

class TrainingWithExercises(TrainingResponse):
    exercises: List[ExerciseResponse] = []